### ISSUE TITLE
Add feature to specify visualCrop: (CropShape) -> CropShape = { it }

### ISSUE THREAD

**Issue by qavan** [tamimattafi/krop#25]
Nice lib❤️
But helpful to have argument like `visualCrop: (CropShape) -> CropShape = { it }` to have "visual crop" of crop area.
One of usecases: cropping image for avatar in social media, when usually you see circle shaped image, but also can view full image(rect). 
![image](https://github.com/user-attachments/assets/95a89e55-c82e-4d30-87ca-5d62983542ee)

**Comment by tamimattafi** [comment]
Hello @qavan!
I'm not sure that I understood correctly what you mean. When you use shapes, you actually can see the full area while you are cropping. But when you press the crop button, you only get the cropped shape. This is a standard behavior for crop libraries.

To show the original picture in social media, it's better to save two versions, one cropped image for the small circular profile picture, and one original image without any cropping, which you will display when the user wants to view the full picture.

**Comment by tamimattafi** [comment]
Here's an example of the current circular crop:
![example-circular](https://github.com/user-attachments/assets/405a38b2-030f-4a4f-832b-27296cc85125)

**Comment by qavan** [comment]
Yeah, visually same, BUT your result is rect image with black background out of circle. I want to have circle visual crop with fully rect image, without black background out of circle.  This is why I want to introduce `visualCrop` feature(I dont know how to better name it)

[Here work as I described, but it not CMP](https://github.com/CanHub/Android-Image-Cropper/blob/71c372b671b06dcba61b4804f1e6cc0e1cb1a66d/cropper/src/main/kotlin/com/canhub/cropper/CropImageOptions.kt#L21)

**Comment by qavan** [comment]
> To show the original picture in social media, it's better to save two versions, one cropped image for the small circular profile picture, and one original image without any cropping, which you will display when the user wants to view the full picture.

It depends

**Comment by tamimattafi** [comment]
@qavan Could you please attach a result bitmap, how it should look like after cropping, without the black background around the circle?

**Comment by tamimattafi** [comment]
@qavan Did I understand correctly, that the result image should have everything inside the cropping grid / square, but show a circular shape while cropping?
Is this example accurate?

![Screenshot_20241126_160753_Krop.jpg](https://github.com/user-attachments/assets/8ddac516-1f35-4667-ad64-ea7aca5bd5ed)

![Screenshot_20241126_160913_Krop.jpg](https://github.com/user-attachments/assets/50b3c99e-4b61-48d7-a316-5bf673e65826)

**Comment by qavan** [comment]
@tamimattafi yes! Exactly!

**Comment by tamimattafi** [comment]
@qavan I understand now, thank you. 

I think adding such a feature isn't hard, but I am not sure if I will find time in the upcoming days, since I'm on vacation.

Please feel free to make a pull request from 0.2.0 branch.

**Comment by vanniktech** [comment]
I'd also be interested in this particular use-case.

**Comment by tamimattafi** [comment]
Hello @qavan, @vanniktech!
Sorry for the delay.

The easiest way I could come up with, to avoid making a lot of changes to the API, is adding a flag to `CropState`.
By changing `clipResultToShape` to `false`, the shape will work only "visually" and have no effect on the end result.
You can set this flag during shape selection, right before cropping, or at the start of the composition.

Is this API convenient for your use-case?
More details can be found in this PR: #45 (A review would be appreciated)

**Comment by tamimattafi** [comment]
These changes will be released in 0.2.0-alpha02
